lib/src/main/java/com/thenewboston/api/common/GetDataSource.kt

```diff
@@ -75,7 +75,11 @@ class GetDataSource @Inject constructor(private val networkClient: NetworkClient
         val urlSuffix = "$validatorsEndpoint/$nodeIdentifier"
         val response = networkClient.defaultClient.get<Validator>(urlSuffix)
 
-        return Outcome.Success(response)
+        return when {
+            response.nodeIdentifier.isEmpty()  ->
+                Outcome.Error("Could not fetch validator with NID $nodeIdentifier", IOException())
```

Comment:
We should leave this to return success outcome. If this was returning a list, we would check for empty list. 

`makeCall` will handle the error from the API inside PrimaryDataSource. Have a look inside `NetworkUtils` where it will catch an error from the API.

Reply:
Ok, so should I remove the test for a single validator from the `GetDataSourceTest` file. Because the test would fail if I remove the condition

Reply:
Yes that's fine to remove the error test case as we don't need it.

Reply:
Hey @tomijaga any progress on this?

Reply:
Yh I forgot to push the changes

Reply:
It's up now